src/windows-emulator/syscalls/memory.cpp

```diff
@@ -280,23 +292,57 @@ namespace syscalls
                                         const emulator_pointer buffer, const ULONG number_of_bytes_to_read,
                                         const emulator_object<ULONG> number_of_bytes_read)
     {
-        number_of_bytes_read.write(0);
+        number_of_bytes_read.try_write(0);
 
         if (process_handle != CURRENT_PROCESS)
         {
             return STATUS_NOT_SUPPORTED;
         }
 
-        std::vector<uint8_t> memory{};
-        memory.resize(number_of_bytes_to_read);
+        uint8_t* memory = new uint8_t[number_of_bytes_to_read];
```

Comment:
```suggestion
        std::vector<uint8_t> memory(number_of_bytes_to_read, 0);
```

that way, you can get rid of the delete below. new/delete is evil 😂